VideoLocker/src/main/java/org/edx/mobile/view/adapters/DiscussionPostsAdapter.java

```diff
@@ -57,60 +68,63 @@ public void render(BaseViewHolder tag, DiscussionThread discussionThread) {
                 }
             } else {
                 icon = FontAwesomeIcons.fa_comments;
-                iconColor = edx_grayscale_neutral_light;
+                iconColor = (discussionThread.isRead() ? edx_grayscale_neutral_dark : edx_brand_primary_base);
             }
             holder.discussionPostTypeIcon.setIcon(icon);
             holder.discussionPostTypeIcon.setIconColor(iconColor);
         }
 
         {
-            String threadTitle = discussionThread.getTitle();
-            holder.discussionPostTitle.setText(threadTitle);
+            final SpannableString threadTitle = new SpannableString(discussionThread.getTitle());
+            if (!discussionThread.isRead()) {
+                CalligraphyTypefaceSpan boldSpan = new CalligraphyTypefaceSpan(openSansSemiBoldFont);
+                threadTitle.setSpan(boldSpan, 0, threadTitle.length(), Spanned.SPAN_EXCLUSIVE_EXCLUSIVE);
+                holder.discussionPostTitle.setText(threadTitle);
+            } else {
+                // Calling toString will truncate the spannable properties applied (if any)
```

Comment:
What if an already created row is being reused that had spans applied to it initially ?